File: Coding/03_Beginner/Full_search/C++/17.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int N;
    cin >> N;
    vector<int> A(N);
    for (int i = 0; i < N; i++) cin >> A[i];

    int cnt = 0;
    for (int i = 0; i < N; i++){
        if (A[i] >= 2){
            bool flag = true;
            for (int j = 2; j < A[i]-1; j++){
                if (A[i] % j == 0){
                    flag = false;
                    break;
                }
            }if (flag) cnt++;
        }
    }
    cout << cnt << endl;
    return 0;
}

File: Coding/03_Beginner/Full_search/Ruby/17.rb
N = gets.to_i
A = gets.split.map(&:to_i)
cnt = 0

A.each do |x|
    if x >= 2
        flag = true
        (2...x-1).each do |j|
            if x % j == 0
                flag = false
                break
            end
        end
    end
    if flag
        cnt += 1
    end
end

puts cnt